File: GoServer/Model/category.go
package Model

type Category struct {
	Id        		string      				`json:"_id" bson:"_id"`
	CategoryName	string 						`json:"categoryName" bson:"categoryName"`
	Brands        	[]string    				`json:"brands" bson:"brands"`
	Colors        	[]string    				`json:"colors" bson:"colors"`	
	PriceRanges 	[]string					`json:"priceRanges" bson:"priceRanges"`
	SellerIds 		[]string					`json:"sellerIds" bson:"sellerIds"`
	// Properties 		bson.M						`json:"properties" bson:"properties"`
	Properties 		map[string][]interface{}	`json:"properties" bson:"properties"`
}

File: src/app/Models/category.ts
export class Category {
    _id: string; 
    categoryName: string;
    brands: string[]; 
    colors: string[]; 
    priceRanges: string[]; 
    sellerIds: string[]; 
    properties: { [key: string]: any[] };

    constructor(
        _id: string,
        categoryName: string,
        brands: string[], 
        colors: string[], 
        priceRanges: string[], 
        sellerIds: string[],
        properties: { [key: string]: any[] },

    ) {
        this._id = _id;
        this.categoryName = categoryName;
        this.brands = brands;
        this.colors = colors; 
        this.priceRanges = priceRanges;
        this.sellerIds = sellerIds;
        this.properties = properties;
    }

}